src/main/java/org/broadinstitute/hellbender/tools/exome/SegmentUtils.java

```diff
@@ -13,292 +15,452 @@
-     *
-     * This will select the overlapping targets from the input collection, for your convenience.
-     *
-     * @param segment -- segment to use in calculating all difference of target CRs from the mean
-     * @param targets -- targets.  Note that this method will use only the targets overlapping the given segment.
-     * @return never {@code null}.  List of doubles of the difference, possibly empty.  Modifiable.
-     */
-    public static List<Double> segmentMeanTargetDifference(final ModeledSegment segment, final TargetCollection<TargetCoverage> targets) {
-        Utils.nonNull(segment, "Can't count targets of null segment.");
-        Utils.nonNull(targets, "Counting targets requires non-null targets collection.");
+    //given breakpointsByContig map, constructs list of untrimmed segments
+    //(i.e., segments that are directly adjacent to each other; segment boundaries at each breakpoint are determined
+    //by whether that breakpoint was a start or an end for the corresponding original target/SNP segment),
+    //then returns only those untrimmed segments that are non-empty
+    private static List<SimpleInterval> constructUntrimmedSegments(final Genome genome,
+                                                                   final SortedMap<String, List<Breakpoint>> breakpointsByContig) {
+        final List<SimpleInterval> segments = new ArrayList<>();
+        for (final String contig : breakpointsByContig.keySet()) {
+            final List<Breakpoint> breakpoints = breakpointsByContig.get(contig);
+            int start = breakpoints.get(0).getStart();
+            for (int i = 1; i < breakpoints.size(); i++) {
+                final Breakpoint breakpoint = breakpoints.get(i);
+                int end;
+                //adjust segment boundaries according to breakpoint type; this prevents, e.g., SNPs that originally
+                //started SNP segments from being stranded
+                switch (breakpoint.getType()) {
+                    case START: end = breakpoint.getSite() - 1;
+                        break;
+                    case END: end = breakpoint.getSite();
+                        break;
+                    default: throw new GATKException.ShouldNeverReachHereException("Unknown breakpoint type in segment union.");
+                }
 
-        final double segMean = segment.getSegmentMeanInCRSpace();
-        final List<TargetCoverage> myTargets = targets.targets(segment);
-        return myTargets.stream().map(t -> ( Math.pow(2, t.getCoverage()) - segMean)).collect(Collectors.toList());
+                if (end < start) {  //this could happen if there are adjacent breakpoints
+                    continue;
+                }
+
+                final SimpleInterval segment = new SimpleInterval(contig, start, end);
+                if (genome.getTargets().targetCount(segment) > 0 || genome.getSNPs().targetCount(segment) > 0) {
```

Comment:
hmm, probably, but i think it's a bit more clear to have the

construct untrimmed -> merge start/ends -> merge middles 

steps broken up explicitly, since performance isn't an issue here.